plugins/auth-backend/src/service/router.ts

```diff
@@ -249,3 +249,22 @@ export function createOriginFilter(
     return allowedOriginPatterns.some(pattern => pattern.match(origin));
   };
 }
+
+/** @internal */
+export function getDefaultBackstageTokenExpiryTime(config: Config) {
+  const processingIntervalKey = 'auth.backstageTokenExpiration';
+
+  if (!config.has(processingIntervalKey)) {
+    return BACKSTAGE_SESSION_EXPIRATION;
+  }
+
+  const duration = readDurationFromConfig(config, {
+    key: processingIntervalKey,
+  });
+  const seconds = Math.max(
+    86400,
+    Math.round(durationToMilliseconds(duration) / 1000),
+  );
```

Comment:
This logic isn't quite right, could you add test cases for all the different expected cases here?

Reply:
Hi @Rugvip, 

As per my understanding, in this logic if I set 3600(60 minutes) instead of 86400 as shown below -
`const seconds = Math.max(3600,Math.round(durationToMilliseconds(duration) / 1000));`
 
then running the unit test - router.tets.ts
```
const config = new ConfigReader({
  app: {
	baseUrl: 'http://example.com/extra-path',
  },
  auth: {
	backstageTokenExpiration: { minutes: 120 }
  }
});
expect(getDefaultBackstageTokenExpiryTime(config)).toBe(7200);
 
```
returns the value passed in app-config.
 
So which means getDefaultBackstageTokenExpiryTime returns the max value.
 
Is this understanding correct, if not can you please elaborate more ?

Reply:
Yep 7200 would be correct, but the test is currently asserting 86400. The minimum value is also gone, we'd expect 2 minutes to give 600

Reply:
Hey @Rugvip 
Thanks for your comment. So just for more clarity, based on the value, we put here **<?>** in below logic
`const seconds = Math.max(**<?>**,Math.round(durationToMilliseconds(duration) / 1000));`

Will get the result accordingly. What value you would suggest to go here ?
If we agree on a value based on previous comments & other reviewers feedback, we can conclude this or let us know if any further suggestion or concern.

Reply:
I think, in short, we need BOTH a Math.min and a Math.max to clamp the value.

Reply:
Hey @freben,

We have pushed the latest patch based on your input, also added multiple testcases. Kindly have a look and let us know in case of any more comments / misinterpretation.